howto_io.md

```diff
@@ -0,0 +1,304 @@
+[numpy.savez_compressed](https://numpy.org/doc/stable/reference/generated/numpy.savez_compressed.html#numpy.savez_compressed)
+to store multiple arrays.
+For [security and portability](#saverestore-using-a-pickle-file), set `allow_pickle=False` unless
+the dtype contains Python objects, which requires pickling.
+
+Masked arrays [can't currently be saved](https://numpy.org/devdocs/reference/generated/numpy.ma.MaskedArray.tofile.html),
+nor can other arbitrary array subclasses.
+
+<a name="32-human-readable"></a>
+### 3.2 Human-readable
+`save` and `savez` create binary files. To write a human-readable file, use
+[numpy.savetxt](https://numpy.org/doc/stable/reference/generated/numpy.savetxt.html).
+The array can only be 1- or 2-dimensional, and there's no `savetxtz` for multiple files.
+
+<a name="33-large-arrays"></a>
+### 3.3 Large arrays
+See [Write or read large arrays](#large_arrays).
+
+<a name="4-read-an-arbitrarily-formatted-binary-file-binary-blob"></a>
+## 4 Read an arbitrarily formatted binary file ("binary blob")
+
+Use a [structured array](https://numpy.org/doc/stable/user/basics.rec.html).
+
+**Example:**
+
+The `.wav` file header is a 44-byte block preceding `data_size` bytes of the
+actual sound data:
+
+```
+chunk_id         "RIFF"
+chunk_size       4-byte unsigned little-endian integer
+format           "WAVE"
+fmt_id           "fmt "
+fmt_size         4-byte unsigned little-endian integer
+audio_fmt        2-byte unsigned little-endian integer
+num_channels     2-byte unsigned little-endian integer
+sample_rate      4-byte unsigned little-endian integer
+byte_rate        4-byte unsigned little-endian integer
+block_align      2-byte unsigned little-endian integer
+bits_per_sample  2-byte unsigned little-endian integer
+data_id          "data"
+data_size        4-byte unsigned little-endian integer
+```
+The `.wav` file header as a NumPy structured dtype:
+```
+wav_header_dtype = np.dtype([
+    ("chunk_id", (bytes, 4)), # flexible-sized scalar type, item size 4
+    ("chunk_size", "<u4"),    # little-endian unsigned 32-bit integer
+    ("format", "S4"),         # 4-byte string
+    ("fmt_id", "S4"),
+    ("fmt_size", "<u4"),
+    ("audio_fmt", "<u2"),     #
+    ("num_channels", "<u2"),  # .. more of the same ...
+    ("sample_rate", "<u4"),   #
+    ("byte_rate", "<u4"),
+    ("block_align", "<u2"),
+    ("bits_per_sample", "<u2"),
+    ("data_id", "S4"),
+    ("data_size", "u4"),
+    #
+    # the sound data itself cannot be represented here:
+    # it does not have a fixed size
+   ])
+
+data = np.fromfile(f,dtype=wave_header_dtype)
+```
+Credit: [Pauli Virtanen](http://scipy-lectures.org/advanced/advanced_numpy/index.html)
+
+<a name="large_arrays">
+
+<a name="5-write-or-read-large-arrays"></a>
+## 5 Write or read large arrays
+
+Arrays too large to fit in memory can be treated like ordinary in-memory arrays using
+[numpy.mmap](https://numpy.org/doc/stable/reference/generated/numpy.memmap.html).

```

Comment:
In light of issue numpy/numpy#16979 maybe add something along the lines of "also useful to reduce the number of data copies made via `large_array[some_slice] = np.load('path/to/small_array', mmap_mode='r')`

Reply:
Great suggestion. Added.